docs/ordinals/nft.md

```diff
@@ -0,0 +1,162 @@
+
+## Inscribe
+
+The following code demonstrates how deploy / inscribe the NFT contract:
+
+```ts
+HashPuzzleNFT.loadArtifact();
+
+const text = "Hello sCrypt and 1Sat Ordinals";
+
+const message = toByteString('secret string', true);
+const hash = sha256(message);
+
+const instance = new HashPuzzleNFT(hash);
+
+const signer = getDefaultSigner();
+await instance.connect(signer);
+
+const inscriptionTx = await instance.inscribeTextNft(text);
+console.log("Inscribed NFT: ", inscriptionTx.id);
+```
+
+The `inscribeTextNft` first inscribes the locking script with the specified text and then deploys the contract.
+
+Among text the inscription can contain many other types of data. Here's how you can conveniently inscribe an image:
+
+```ts
+// ...
+
+const bb = readFileSync(join(__dirname, "..", "..", "logo.png")).toString("base64");
+
+const tx = await instance.inscribeImageNft(bb, ContentType.PNG);
+console.log("Inscribed NFT: ", tx.id);
+```
+
+In fact the data type can be arbitrary. It only depends on the Ordinals wallet you're using to support that data type.
+
+```ts
+const tx = await instance.inscribe({
+  content: `your content in hex`,
+  contentType: `your contentType`,
+});
+console.log("Inscribed NFT: ", tx.id);
+```
+
+The value `contentType` must be a MIME-type string. The [`ContentType`](https://github.com/sCrypt-Inc/scrypt-ord/blob/master/src/contentType.ts) object contains common MIME-types.
+
+## Transfer
+
+You can easily transfer a deployed NFT to an Ordinals address by passing a `transfer` value via the method call parameters.
+
+```ts
+// ... deploy code from above
+
+const { tx: transferTx } = await instance.methods.unlock(
+    message, 
+    {
+        transfer: new OneSatNFTP2PKH(
+                Addr(recipientAddress.toByteString())
+            ),
+    }
+);
+
+console.log("Transferred NFT: ", transferTx.id);
+```
+
+### Transfer Existing NFT
+
+Suppose you would like to transfer an existing NFT that was already inscribed in the past, which is typically locked using a `P2PKH` lock.
+You can fetch all the needed data for the transfer by either using `fromUTXO` or `getLatestInstance`. The former takes the deployed NFT's current UTXO, while the latter takes the NFT's [origin](https://docs.1satordinals.com/readme/terms#origin).
+
+If the deployed NFT is locked using a regular `P2PKH` you may unlock it like the following:
+
+```ts
+const outpoint = '036718e5c603169b9981a55f276adfa7b5d024616ac95e048b05a81258ea2388_0';
+
+// Create a P2PKH object from a UTXO
+const utxo: UTXO = OneSatApis.fetchUTXOByOutpoint(outpoint);
+const p2pkh = OneSatNFTP2PKH.fromUTXO(utxo);
+// Alternatively, create a P2PKH from an origin
+const p2pkh = await OneSatNFTP2PKH.getLatestInstance(outpoint);
+
+// Construct recipient smart contract
+const message = toByteString('super secret', true);
+const hash = sha256(message);
+const recipient = new HashPuzzleNFT(hash);
+await recipient.connect(getDefaultSigner());
```

Comment:
Do we need connect here, if we do not further call the recipient contract?

Reply:
Yes, call happens under the hood since we pass this instance in the method call params, I think.

Reply:
NO, no need to connect if no further call .